### DISCUSSION TITLE
Send Text Directly to Receiver Not Via ejabberd

### DISCUSSION THREAD

**Discussion by n4b7l** [processone/ejabberd#4159] [General]
I know how STUN/TURN works to get IP address of the receiver to make calls. Wondering whether it's possible for text messages as well? I want clients can directly send text messages each other but not via ejabberd server. Is it possible at all?

**Comment by licaon-kter** [comment]
Like https://xmpp.org/extensions/xep-0174.html ?

**Reply by mremond** [reply]
I think what he has in mind is more about the ability to open a side-channel chat with messages not going through the server.

**Comment by mremond** [comment]
In your case, you plan to use XMPP for negociation of how to establish a connection between peers.

Once all party agree that can connect out of band, i.e. outside of the XMPP stream.

If your protocol for direct communication is based on UDP, you should be able to transfer whatever you want over TURN.
